doc/source/blueprint_cce_vault.rst

```diff
@@ -0,0 +1,592 @@
+
+Since in our example a dedicated service account with token is being
+periodically rotated using `client JWT as reviewer JWT
+<https://www.vaultproject.io/docs/auth/kubernetes#use-the-vault-client-s-jwt-as-the-reviewer-jwt>`_
+can be used.
+
+Access rules for Vault
+======================
+
+Once Auth plugin is enabled CCE workloads are able to authenticate to Vault,
+but they can do nothing. It is now necessary to establish further level of
+authorization and let particular service account of CCE to get access to
+secrets in vault.
+
+For the scope of the use case we are going to allow zookeeper service account
+from zookeeper namespace to get access to the TLS secrets stored in vault (KV
+storage). For that a policy that gives a read only access to the /tls/zk* and
+/tls/ca paths is being created.
+
+.. code-block:: shell
+
+   $ vault policy write tls-zk-ro - <<EOF
+   path "secret/data/tls/zk_*" {capabilities = ["read"] }
+   path "secret/data/tls/ca" {capabilities = ["read"] }
+   path "secret/metadata/tls/zk_*" {capabilities = ["read"] }
+   path "secret/metadata/tls/ca" {capabilities = ["read"] }
+   EOF
+
+Next granting the policy to the particular requestor (zookeeper
+service account in zookeeper namespace) must be done.
+
+.. code-block:: shell
+
+   $ vault write auth/kubernetes_cce1/role/zookeeper \
+       bound_service_account_names=zookeeper \
+       bound_service_account_namespaces=zookeeper \
+       policies=tls-zk-ro \
+       ttl=2h
+
+With this done token of the service account zookeeper in the zookeeper
+namespace would be able to access to the vault for reading secrets located
+under `/secret/tls` path. And since it is higly recommended to follow the least
+required privilege principle only read only access to the TLS data is granted.
+A time to live of 3 hours is being used here meaning that once application
+authorize to Vault the token it gets can be used during next 2 hours. This need
+to be carefully aligned with the time to live or the service account token to
+minimize their overlap. It is advised to keep it relatively short.
+
+This is one the most sensitive steps in the whole configuration, since it can
+not be generally excluded that the applications deployed in the Kubernetes find
+escape or be compromised. For that case it is required to strictly limit amount
+of secrets accessor will be able to read.
+
+Populating secrets in Vault
+===========================
+
+Within Vault there are 2 possibilities to access TLS certificates:
+
+* Store certificate data in the `KeyValue store
+  <https://www.vaultproject.io/docs/secrets/kv/kv-v2>`_
+
+* Use `PKI secrets engine <https://www.vaultproject.io/docs/secrets/pki>`_ to
+  issue certificates
+
+Vault enables users not only to store TLS certificates data in the KV store,
+but also to create and  revoke them. To keep this tutorial simple enough we are
+not going to do this and only upload generated certificates into the KV store.
```

Comment:
would be good to tell the readers that ``` @zk_server.crt ``` means it expects a file named zk_server.crt on the filesystem from where the vault command is executed